### DISCUSSION TITLE
Remove a repo from aur_repos after package removal

### DISCUSSION THREAD

**Discussion by kriansa** [actionless/pikaur#759] [Q&A]
I haven't found any way of doing that through `pikaur` interface.

I'm not sure if I understand why this works like that. Is there any advantage to keeping packages under that repo even past their removal?

Or are there any plans to add support for that in the long run?

**Comment by actionless** [comment]
i didn't really understood the question

**Comment by kriansa** [comment]
Oh, my apologies, the terminology is really confusing.

When we install a new package from AUR, Pikaur first verify if we have an AUR git repository present on `~/.local/share/pikaur/aur_repos`, and if not, it will fetch and clone that AUR to there.

However, if that package gets removed, the AUR git repo will stay there, and in the eventuality of a new reinstall, the user will be prompted to approve the diff between the existing and the what has changed on that AUR.

As of now, I couldn't find anything on Pikaur's interface that removes the "AUR repo" altogether, and one must do it manually if they so desire. If that thing exists, please point me out to how to do it. If not, is there a planned feature for also cleaning that up, or are there other design decisions for keeping them there that I'm not aware about?

Thanks!

**Reply by actionless** [reply]
yeah, the idea is that the fact that you removing the package is not removing the fact you reviewed it, thus that information is preserved

aur repo, unlike src repo of a package itself - weights only few kilobytes, so i was never considering storing them all an issue

**Reply by kriansa** [reply]
Got it, thanks for the reply!

To clarify a bit, I'm not at all concerned about space for their storage, but the fact that I might install a package for testing purposes, remove it and long later I could re-install it and find it odd that the review wouldn't make a lot of sense since I had no idea I once installed it.

Thanks for the all the awesome work on the project!

**Reply by actionless** [reply]
it sounds like quite an edge case to me, but if it feels important to you - feel free to send a PR adding new --key to -R command, for optionally removing that dir

**Reply by actionless** [reply]
you could use this code (for adding new behavior to -Sc command) as an example: https://github.com/actionless/pikaur/blob/master/pikaur/pkg_cache_cli.py

**Reply by kriansa** [reply]
Oh, it's fine how it is today, don't worry. I was really just curious if this was already on your mind or what led you to take the decision.

For now, that doesn't seem as bad as I might have explained, I'll see how it goes over time. There are complexities involved in removing the AUR repo, especially because there's no 1:1 relationship between a package and a repo, and there are also GPG keys that should be removed when not needed.

**Reply by actionless** [reply]
pikaur was never adding any keys at first place, so obviously don't need to remove them :)
